ocean_lib/web3_internal/contract_base.py

```diff
@@ -55,13 +55,13 @@ def __init__(self, address: Optional[str], abi_path=None):
         assert abi_path, f"abi_path is required, got {abi_path}"
 
         self.w3 = Web3Provider.get_web3()
-        self.contract_concise = get_concise_contract(self.w3, self.name, address)
+        self.contract_caller = get_contract_caller(self.w3, self.name, address)
```

Comment:
do we still need this assignment? If the caller can be accessed using `self.contract.caller` I don't see the point in assigning another instance variable :)

Reply:
This function `get_contract_caller` returns a `ContractCaller` instance with the specific abi, web3 instance and address, as contract_concise was before with `get_concise_contract`. And after I use this variable `contract_caller` in `dispenser`, too. Maybe we can replace with `contract_caller` wherever is possible.
Other than that, if I remove the assignment, pytest hangs (I waited some couple of minutes and nothing happened).

![](https://files.slack.com/files-pri/T0D6J2S9L-F027G44C4BS/image.png)

Reply:
UPDATE: @calina-c, I removed the assignment for contract_caller and that `get_contract_caller`, too.